docs/content/guide/forms.ngdoc

```diff
@@ -44,7 +44,9 @@ for other directives to augment its behavior.
           };
 
           $scope.reset = function() {
-            $scope.user = angular.copy($scope.master);
+            //$scope.user = angular.copy($scope.master);
+            $scope.master = {};
+            $scope.user = {};
```

Comment:
I agree. The idea is that the `$scope.user` object is modified when the input control values are changed. Clicking reset returns it to the "saved" value, which is in `$scope.master`.